### PR TITLE
Multiple bug fixes in the data providers

### LINKED ISSUE #195: Better data providers for sanitizers
None of the projects use sanitizers. That looks suspicious:

https://github.com/artem-smotrakov/fosstars-rating-core/blob/analyse-ratings/src/main/jupyter/oss/security/SecurityRatingAnalysis.ipynb

Let's look into it.

### LINKED ISSUE #203: ScansForVulnerableDependencies should not be indecisive
The provider returns an unknown value if no known dependency scanner is used:

https://github.com/SAP/fosstars-rating-core/blob/650f483c81d3c39325625930ee91fd8d3f3df0fc/src/main/java/com/sap/sgs/phosphor/fosstars/data/github/ScansForVulnerableDependencies.java#L52

That results to a lower confidence of the rating. It would be better if the provider just returned false.

### LINKED ISSUE #204: UsesGithubForDevelopment should expect empty descriptions
`UsesGithubForDevelopment` tries to read a description of a repository but doesn't expect that it may be null. Therefore, it may results to the following NPE:

```
java.lang.NullPointerException
        at com.sap.sgs.phosphor.fosstars.data.github.UsesGithubForDevelopment.lambda$static$1(UsesGithubForDevelopment.java:36)
        at com.sap.sgs.phosphor.fosstars.data.github.UsesGithubForDevelopment.lambda$usesGitHubForDevelopment$5(UsesGithubForDevelopment.java:108)
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
        at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.util.stream.ReferencePipeline.reduce(ReferencePipeline.java:474)
        at com.sap.sgs.phosphor.fosstars.data.github.UsesGithubForDevelopment.usesGitHubForDevelopment(UsesGithubForDevelopment.java:109)
        at com.sap.sgs.phosphor.fosstars.data.github.UsesGithubForDevelopment.usesGithubForDevelopment(UsesGithubForDevelopment.java:90)
        at com.sap.sgs.phosphor.fosstars.data.github.UsesGithubForDevelopment.fetchValueFor(UsesGithubForDevelopment.java:76)
```

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Here is a list of updates:

- Updated `ScansForVulnerableDependencies` data provider to return false if no scanner found. This closes #203.
- Improved `UsesSanitizers` data provider. This closes #195.
- Updated `UsesGithubForDevelopment` data provider to expect empty descriptions. This closes #204.

### CODE REVIEW

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/ScansForVulnerableDependencies.java [hunk 1]
```diff
@@ -1,26 +1,28 @@
...
 import java.util.List;
 
 /**
- * This data provider tries to fill out the
- * {@link
- * com.sap.sgs.phosphor.fosstars.model.feature.oss.OssFeatures#SCANS_FOR_VULNERABLE_DEPENDENCIES}.
- * feature.
- * It is based on a number of data providers:
+ * This data provider tries to fill out the {@link OssFeatures#SCANS_FOR_VULNERABLE_DEPENDENCIES}.
+ * feature. It is based on the following data providers:
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/data/github/ScansForVulnerableDependencies.java:16]: data provider?

Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/github/ScansForVulnerableDependencies.java:16]: The list currently contains one data provider but it's still a list.

File: src/main/java/com/sap/sgs/phosphor/fosstars/data/github/ScansForVulnerableDependencies.java [hunk 2]
```diff
@@ -43,12 +45,12 @@ protected Feature supportedFeature() {
   @Override
   protected Value fetchValueFor(GitHubProject project) throws IOException {
     for (CachedSingleFeatureGitHubDataProvider provider : providers) {
-      Value value = provider.fetchValueFor(project);
-      if (!value.isUnknown() && Boolean.TRUE.equals(value.get())) {
+      Value<Boolean> value = provider.fetchValueFor(project);
+      if (value.orElse(FALSE)) {
```
Comment (sourabhsparkala) [inline src/main/java/com/sap/sgs/phosphor/fosstars/data/github/ScansForVulnerableDependencies.java:49]: May be this entire thing can be written in one line
- Like `return SCANS_FOR_VULNERABLE_DEPENDENCIES.value(value.orElse(FALSE))` ?

Comment (artem-smotrakov) [reply src/main/java/com/sap/sgs/phosphor/fosstars/data/github/ScansForVulnerableDependencies.java:49]: If so, then only the first data provider will be called which is wrong.

**Comment by artem-smotrakov** [comment]
Thanks for the review @sourabhsparkala !

**Pull request merged**